bin/write_viral_gff.py

```diff
@@ -21,136 +20,213 @@ def get_ena_contig_mapping(ena_contig_file):
     return ena_mapping
 
 
-def write_gff(virify_files, sample_prefix, ena_mapping=None):
-    contigs = {}
-    if ena_mapping:
-        ena_assembly_accession = list(ena_mapping.values())[0].split(".")[0]
-        output_filename = f"{ena_assembly_accession}_virify.gff"
-    else:
-        output_filename = f"{sample_prefix}_virify.gff"
-    with open(output_filename, "w") as gff:
-        gff.write(f"##gff-version 3\n")
-        #   annotation type specify this is virify confidence level
-        for vfile in virify_files:
-            virify_df = pd.read_csv(vfile, sep="\t")
-            filtered_df = virify_df[virify_df["Best_hit"] != "No hit"]
-            #   set lowest possible start and highest possible end
-            for index, row in filtered_df.iterrows():
-                if not row["Contig"] in contigs:
-                    contigs[row["Contig"]] = {"start": row["Start"], "end": row["End"]}
+def collect_annot(virify_files, sample_prefix, ena_mapping=None):
+    ## Parsing the annotation files
+    contig_protID={}
+    prot_data={}
+    contig_data={}
+    for vfile in virify_files:
+        with open(vfile,'r') as input_table:
+            next(input_table)
+            for line in input_table:
+                line_l=line.rstrip().split('\t')
```

Comment:
it's easier to read like this:
```suggestion
                contig, cds_id, start, end, direction, best_hit = line.rstrip().split('\t')
```

Reply:
We cannot do it that way because the table has 6 fields when `best_hit=='No hit'` and 7 fields when `best _hit != 'No hit'`

Reply:
Alright